lib/tapioca/sorbet_ext/name_patch.rb

```diff
@@ -6,10 +6,11 @@ module Types
     class Simple
       module NamePatch
         def name
-          @name ||= ::Tapioca::Reflection.name_of(@raw_type).freeze
+          @name ||= ::Tapioca::Reflection.qualified_name_of(@raw_type).freeze
         end
       end
 
+      T::Types::Simple::Private::Pool.instance_variable_set(:@cache, ObjectSpace::WeakMap.new)
```

Comment:
The problem apparently was that existing instances of `T::Types::Simple` has memoized their names in the `@name` ivar, so they were holding on to that even after we prepend the patch.

The fix is to use a different memoizing ivar in the prepended module, so that they all start fresh. However, that causes an infinite loop descending into the `qualified_name_of` method on `Reflection` since the method has a signature which tries to do the same thing recursively.

Thus, we need a copy of the `qualified_name_of` method here without a signature to fix everything:

```suggestion
          @qualified_name ||= qualified_name_of(@raw_type).freeze
        end

        private

        NAME_METHOD = Module.instance_method(:name)

        def qualified_name_of(constant)
          name = NAME_METHOD.bind(constant).call
          return if name.nil?

          if name.start_with?("::")
            name
          else
            "::#{name}"
          end
        end
      end
```

Reply:
This fixed the `T.nilable` conversions. However, there are still some types that are unaffected by the patch in the tests. Not sure if it's a blocker for merging this PR.